lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -0,0 +1,302 @@
+
+            templateSelector: '#field_text-tpl',
+
+            events: {
+                'change input': 'saveValue',
+            },
+
+            initialize: function (options) {
+                this._super(options);
+                _.bindAll(this, 'render', 'fieldValue', 'updateValueInField', 'saveValue');
+                this.listenTo(this.model, "change:" + this.options.valueAttribute, this.updateValueInField);
+            },
+
+            render: function () {
+                this.$el.html(this.template({
+                    title: this.options.title,
+                    value: this.modelValue(),
+                    message: this.helpMessage,
+                }));
+                return this;
+            },
+
+            fieldValue: function () {
+                return this.$('.account-settings-field-value input').val();
+            },
+
+            updateValueInField: function () {
+                this.$('.account-settings-field-value input').val(this.modelValue() || '');
+            },
+
+            saveValue: function (event) {
+                var attributes = {};
+                attributes[this.options.valueAttribute] = this.fieldValue();
+                this.saveAttributes(attributes);
+
+            },
+        });
+
+        AccountSettingsFieldViews.EmailFieldView = AccountSettingsFieldViews.TextFieldView.extend({
+
+            successMessage: function() {
+                return this.indicators['success'] + interpolate_text(
+                    gettext('Complete you email change by clicking the confirmation link emailed to {new_email_address}.'),
+                    {'new_email_address': this.fieldValue()}
+                );
+            },
+        });
+
+        AccountSettingsFieldViews.DropdownFieldView = AccountSettingsFieldViews.FieldView.extend({
+
+            templateSelector: '#field_dropdown-tpl',
+
+            events: {
+                'change select': 'saveValue',
+            },
+
+            initialize: function (options) {
+                this._super(options);
+                _.bindAll(this, 'render', 'fieldValue', 'updateValueInField', 'saveValue');
+                this.listenTo(this.model, "change:" + this.options.valueAttribute, this.updateValueInField);
+            },
+
+            render: function () {
+                this.$el.html(this.template({
+                    title: this.options.title,
+                    required: this.options.required,
+                    selectOptions: this.options.options,
+                    message: this.helpMessage
+                }));
+                this.updateValueInField();
+                return this;
+            },
+
+            fieldValue: function () {
+                return this.$('.account-settings-field-value select').val();
+            },
+
+            updateValueInField: function () {
+                this.$('.account-settings-field-value select').val(this.modelValue() || "");
```

Comment:
Using ``||`` means that any falsey value will not be included. In particular, this is a problem if the value is 0.